### PR TITLE
Added HCI_EXT_PacketErrorRate command.

### PR DESCRIPTION
**Pull request by hoarf**

### CODE REVIEW

File: hci/command/commands/hci_ext_per.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from struct import pack, unpack
+
+from ..command_packet import CommandPacket
+from enum import IntEnum
+
+
+class HCI_EX_PER(CommandPacket):
```
Comment (acburigo) [inline hci/command/commands/hci_ext_per.py]: The class name is given by the `TI_BLE_Vendor_Specific_HCI_Guide.pdf` file.

**Review by acburigo — CHANGES_REQUESTED** [review]
You need to add your command to autocast and init.

Please, refer to any pull request adding a new command so you have an idea of how to add a new command to this library.

**Comment by hoarf** [comment]
I made the requested changes, please review it again.

File: hci/command/commands/__init__.py [hunk 2]
```diff
@@ -11,3 +11,4 @@
 from .hci_reset import HCI_Reset
 from .util_build_revision import UTIL_BuildRevision
 from .util_force_boot import UTIL_ForceBoot
+from .hci_packet_error_rate_cmd import HCI_EXT_PacketErrorRateCmd
```
Comment (acburigo) [inline hci/command/commands/__init__.py]: You can remove `cmd` from the file name and `Cmd` from the class name.

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from struct import pack, unpack
+
+from ..command_packet import CommandPacket
+from enum import IntEnum
+
+
+class HCI_EXT_PacketErrorRateCmd(CommandPacket):
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: You can remove `cmd` from the file name and `Cmd` from the class name.

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+from struct import pack, unpack
+
+from ..command_packet import CommandPacket
+from enum import IntEnum
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: `enum` should be imported with the top group of imports.
First group contains python modules, second group contains project modules.

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+class HCI_EXT_PacketErrorRateCmd(CommandPacket):
+
+    class Command(IntEnum):
+        HCI_EXT_PER_RESET = 0x00
+        HCI_EXT_PER_READ = 0x01
+
+    def __init__(self, conn_handle, command):
+        super().__init__(
+            CommandPacket.OpCode.HCI_EX_PER,
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: This OpCode does not exist. Please, test your command before sending a pull request. You can test it the same way we show on our README.md example.

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    def conn_handle(self):
+        OFFSET, SIZE_OCTETS = 4, 2
+        data = self._get_data(OFFSET, SIZE_OCTETS)
+        return unpack('<H', data)[0]
+
+    @property
+    def command(self):
+        OFFSET, SIZE_OCTETS = 6, 2
+        data = self._get_data(OFFSET, SIZE_OCTETS)
+        return unpack('<H', data)[0]
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: `command` parameter is a single octet, so `SIZE_OCTETS = 1` and `unpack('<B', data)[0]`.

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    def __str__(self):
+        return super().__str__() + '\n' + '\n'.join([
+            'Connection Handle: {} ({})',
+            'Command: {} ({})',
+        ]).format(
+            hex(self.conn_handle),
+            int(self.conn_handle),
+            hex(self.command),
+            int(self.command),
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: Translate `command` using the `enum` you created.
For example: `HCI_EXT_PacketErrorRateCmd.Command(self.command).name`

File: hci/command/commands/hci_ext_packet_error_rate_cmd.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        )
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate_cmd.py]: File ends with a new line character.

**Review by acburigo — CHANGES_REQUESTED** [review]

**Comment by hoarf** [comment]
all changes requested were made and the pr is passing the test describing in the readme

File: hci/command/commands/__init__.py [hunk 9]
```diff
@@ -11,3 +11,4 @@
 from .hci_reset import HCI_Reset
 from .util_build_revision import UTIL_BuildRevision
 from .util_force_boot import UTIL_ForceBoot
+from .hci_ext_packet_error_rate import HCI_EXT_PacketErrorRate
```
Comment (acburigo) [inline hci/command/commands/__init__.py]: This is listed in alphabetic order.

File: hci/command/autocast.py [hunk 10]
```diff
@@ -42,6 +42,9 @@ def _autocast(pkt):
 
         OpCode.UTIL_FORCE_BOOT:
             UTIL_ForceBoot,
+
+        OpCode.HCI_EXTENSION_PACKET_ERROR_RATE:
+            HCI_EXT_PacketErrorRate,
     }
```
Comment (acburigo) [inline hci/command/autocast.py]: Same order as `__init__.py`.

File: hci/command/commands/hci_ext_packet_error_rate.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+    def __init__(self, conn_handle, command):
+        super().__init__(
+            CommandPacket.OpCode.HCI_EXTENSION_PACKET_ERROR_RATE,
+            HCI_EXT_PacketErrorRate._params_to_binary(conn_handle, command)
+        )
+
+    @staticmethod
+    def _params_to_binary(conn_handle, client_rx_mtu):
+        return pack('<HH', conn_handle, client_rx_mtu)
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate.py]: `<HB`

File: hci/command/commands/hci_ext_packet_error_rate.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+    def __str__(self):
+        return super().__str__() + '\n' + '\n'.join([
+            'Connection Handle: {} ({})',
+            'Command: {} ({})',
+        ]).format(
+            hex(self.conn_handle),
+            int(self.conn_handle),
+            HCI_EXT_PacketErrorRate.Command(self.command).name,
+            HCI_EXT_PacketErrorRate.Command(self.command).value,
```
Comment (acburigo) [inline hci/command/commands/hci_ext_packet_error_rate.py]: First, comes the hexadecimal value, then comes the name (between parenthesis). Check other commands if you are in doubt.

**Review by acburigo — CHANGES_REQUESTED** [review]

**Comment by hoarf** [comment]
fixed the ordering, the 'HH' issue and the packet name

**Review by acburigo — APPROVED** [review]

**Pull request merged**